libs/providers/vmware.py

```
@@ -1290,20 +1290,40 @@ def clone_vm(
+            LOGGER.info(f"Using configured resource pool: {configured_pool_name}")
+        elif host_obj:
+            # When a target host is specified, use its parent compute resource's pool to ensure compatibility
+            relocate_spec.pool = host_obj.parent.resourcePool
+            LOGGER.info(f"Using resource pool from target host's compute resource: {host_obj.parent.name}")
+        elif source_vm.resourcePool:
+            # Use source VM's resource pool if available
+            relocate_spec.pool = source_vm.resourcePool
+            LOGGER.info("Using source VM's resource pool")
+        else:
+            # If source is a template with no pool, find a suitable one from the cluster
             container = self.view_manager.CreateContainerView(self.content.rootFolder, [vim.ComputeResource], True)
```

Ruff (0.15.20)

[warning] Boolean positional value in function call

(FBT003)